### ISSUE TITLE
Do not define internal methods like `init` directly on the entity prototype

### ISSUE THREAD

**Issue by hong-duc** [mikro-orm/mikro-orm#506]
**Describe the bug**
I didn't know mikro-orm will add a method name `init` to entity, so in my entity class I also add a method name `init`, when I create the entity with `new Backup()` it throw `UnhandledPromiseRejectionWarning: ValidationError: Entity Backup is not managed. An entity is managed if its fetched from the database or registered as new through EntityManager.persist()`

**Stack trace**
```
(node:16010) UnhandledPromiseRejectionWarning: ValidationError: Entity Backup is not managed. An entity is managed if its fetched from the database or registered as new through EntityManager.persist()
    at Function.entityNotManaged (/home/hongduc/Project/backup/backup-service/node_modules/mikro-orm/dist/utils/ValidationError.js:62:16)
    at Function.init (/home/hongduc/Project/backup/backup-service/node_modules/mikro-orm/dist/entity/EntityHelper.js:14:43)
    at Backup.prototype.init (/home/hongduc/Project/backup/backup-service/node_modules/mikro-orm/dist/entity/EntityHelper.js:95:33)
    at new Backup (/home/hongduc/Project/backup/backup-service/dist/domain/model/backup/Backup.js:21:14)
    at BackupService.createBackup (/home/hongduc/Project/backup/backup-service/dist/application/backup.service.js:33:24)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:16010) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:16010) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
...
```

**To Reproduce**
1. write a entity class with method `init` in it
2. create entity with `new`

**Expected behavior**
Not throw any error

**Additional context**
It easy to solve by just the method name on my side, it just that this method is followed by my team standard and it has used in multiple class now, so I hope that mikro-orm team will consider change method name on your side. Thanks

**Versions**

| Dependency | Version |
| - | - |
| node | v10.19.0 |
| typescript | ^3.6.3 |
| mikro-orm | ^3.4.1 |
| mongodb | ^3.5.5 |

**Comment by B4nan** [comment]
We could definitely move the method under `Entity.prototype.__internal` and map to that via `wrap()` method. And instead of the `WrappedEntity` interface we could have a `BaseEntity` for classic extension. Will think about this in v4, as it might end up being breaking change (which renaming the method would definitely be anyway).

**Comment by B4nan** [comment]
Thinking about this more, and I will probably go for a bit more breaking (but much cleaner) approach:

- all the public (not prefixed) methods will be part of a new class (let's say instance of EntityHelper that is aware of the entity, under `entity.__helper`)
- only properties prefixed with `__` will be part of the entity prototype
- this will have one breaking change for `wrap()` as it won't return the entity anymore, but rather the helper instance, so no access to entity properties won't be possible on the return value
- instead of interface merging with `WrappedEntity` it will be required to actually extend it (as it will be converted to class)
- we might be able to use the same class for the helper instance and for extension

**Comment by B4nan** [comment]
Closing as implemented via #531, which is now merged in v4 (`dev` branch). Subscribe here for updates #527.
